src/resources/listings.js

```diff
@@ -99,6 +108,63 @@ class Listings extends ResourceBase {
     return range(0, Number(listingsLength))
   }
 
+  async search(query) {
+    // Issues an API call to the bridge server for searching listings.
+    // Args:
+    //   query(SearchQuery): query object.
+    // Returns:
+    //   list(int): listing ids that match the query.
+    if (query.rawQuery == '') {
+      return []
+    }
+
+    // Make the search API call to the bridge server.
+    const url = encodeURI(
+      appendSlash(this.indexingServerUrl) + 'api/search/listings?query=' + query.rawQuery)
```

Comment:
Just FYI javascript also has string interpolation, which I like. E.g. this could also be written

```javascript
appendSlash(`${this.indexingServerUrl}api/search/listings?query=${query.rawQuery}`)
```